### PR TITLE
Fix for Button ADC value of Zero

### PR DESCRIPTION
**Pull request by fraser125**
This addresses an issue that I was having when the Button ADC value was Zero.

### CODE REVIEW

**Comment by rlogiacco** [comment]
Please explain what type of issue were you having and how these would improve the situation as this is unclear to me

File: AnalogButtons.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ void AnalogButtons::check() {
 	// In case this function gets called very frequently avoid sampling the analog pin too often: max frequency is 50Hz
 	if (millis() - time > 120) {
 		time = millis();
-		uint16_t reading = analogRead(pin);
```
Comment (rlogiacco) [inline AnalogButtons.cpp]: reading can **never** be negative, this doesn't make any sense, not even within the context of #5

File: AnalogButtons.h [hunk 2]
```diff
@@ -35,7 +35,7 @@
 
 class Button {
 public:
-	uint16_t value;
+	int16_t value;
```
Comment (rlogiacco) [inline AnalogButtons.h]: reading can never be negative, this doesn't make any sense, not even within the context of #5

**Comment by fraser125** [comment]
I agree that the reading can never be less than zero. 
The reason this "makes sense", **I'm still not saying it's the right solution**. 
The if statement calculates "value - margin" since the result of this calculation doesn't have it's own variable type the compiler makes an assumption based on the other variables in the 'if' statement, in order to allow the result of this calculation to be negative the other values have to support negative numbers. 

For example the following achieves the same thing as my proposed fix for this edge case. I'm not thrilled with the amount of casting that happens in this line but the results are correct based on my testing, it also shows where the actual issue occurs vs. being in far off places in the code where it's not clear what the actual use case is.
`if ((int)reading >= (int)buttons[i].value - margin && (int)reading <= (int)buttons[i].value + margin) {`

I did test your proposed 'if' statement and it fired on the wrong button.
The relevant test data is 
`if ((reading >= buttons[i].value - margin || reading <= margin) && reading <= buttons[i].value + margin) {`

> reading = 0
> buttons[i].value - margin =629
> margin = 10
> buttons[i].value + margin = 649
> Inside If
> Select click

It should of been Right click, Select is the first Button that is added to the AnalogButton object. Commenting out the Add for the Select button causes the Left button to fire (next one added).

`if ((reading >= buttons[i].value - margin || reading <= margin) && reading <= buttons[i].value + margin) {`

> reading = 0
> buttons[i].value - margin =398
> margin = 10
> buttons[i].value + margin = 418
> Inside If
> Left click

At this time I would like to cancel this pull request and create a new one based on the 'if' statement above with the casting. It actually works better than my original proposal and is much more clear on where the issue is.

**Pull request closed without merging**